src/runtime/linux/auxvec.rs

```diff
@@ -0,0 +1,86 @@
+//! Reads /proc/self/auxv on Linux systems
+
+use std::prelude::v1::*;
+use std::slice;
+use std::mem;
+use std::fmt;
+
+/// Simple abstraction for the ELF Auxiliary Vector
+///
+/// the elf.h provide the layout of the single entry as auxv_t.
+/// The desugared version is a usize tag followed by a union with
+/// the same storage size.
+pub struct AuxVec {
+    raw: Vec<usize>,
```

Comment:
I think we should avoid the `Vec` here. Can't we just use a `[usize; 64]` + a length?

Reply:
I could just copy the hwcaps when present maybe?

Reply:
I think that would be better. 

I know we are already doing a lot of syscals by reading the file, and that allocating the vector should not matter much, but I was hoping that if we managed to add support for `libc::getauxval` in the future we could reuse at least some of this for that.